baseframe/forms/fields.py

```diff
@@ -503,3 +503,19 @@ def process(self, *args, **kwargs):
         self.form.csrf_enabled = False
         del self.form.csrf_token
         return retval
+
+class MapMarkerField(wtforms.Field):
+    """
+    Adds a map widget that allows the user to search for a location and drag the marker on the map.
+    See the add_map_marker_widget macro in forms.html.
+    """
+    def __init__(self, label='', validators=None, attr='map', map_id='map', latitude_id='', longitude_id='', **kwargs):
```

Comment:
Don't accept: `map_id`, `latitude_id`, `longitude_id`. Autogenerate all of them.

Also, you need to supply a widget that will be used if there is no JS. Create a new widget class that renders two text fields for lat and lon.